modules/base/include/modules/base/processors/filecache.h

```diff
@@ -138,6 +105,138 @@ void updateFilenameFilters(const DataReaderFactory& rf, const DataWriterFactory&
 
 }  // namespace detail
 
+class IVW_MODULE_BASE_API CacheBase : public Processor, public ProcessorNetworkEvaluationObserver {
+public:
+    explicit CacheBase(InviwoApplication* app);
+    virtual bool isConnectionActive(Inport* inport, Outport* outport) const override;
+    virtual void onProcessorNetworkEvaluationBegin() override;
+
+    virtual bool hasCache(std::string_view key) = 0;
+
+protected:
+    void writeXML() const;
+
+    BoolProperty enabled_;
+    DirectoryProperty cacheDir_;
+    DirectoryProperty refDir_;
+    StringProperty currentKey_;
+
+    bool isCached_ = false;
+    std::string key_;
+    std::pmr::string xml_;
+};
+
+template <typename DataType>
+struct ReaderWriter {
+    explicit ReaderWriter(InviwoApplication* app)
+        : extensions{"readerWriter", "Data Reader And Writer"}
+        , rf{app->getDataReaderFactory()}
+        , wf{app->getDataWriterFactory()} {
+
+        detail::updateFilenameFilters<DataType>(*rf, *wf, extensions);
+        extensions.setCurrentStateAsDefault();
+    }
+
+    std::unique_ptr<DataReaderType<DataType>> getReader() {
+        if (extensions.empty()) return nullptr;
+
+        const auto& sext = extensions.getSelectedValue().extension_;
+        return rf->template getReaderForTypeAndExtension<DataType>(sext);
+    }
+    std::unique_ptr<DataWriterType<DataType>> getWriter() {
+        if (extensions.empty()) return nullptr;
+
+        const auto& sext = extensions.getSelectedValue().extension_;
+        return wf->template getWriterForTypeAndExtension<DataType>(sext);
+    }
+
+    OptionProperty<FileExtension> extensions;
+
+private:
+    const DataReaderFactory* rf;
+    const DataWriterFactory* wf;
+};
+
+template <typename DataType>
+struct RamCache {
+    RamCache()
+        : size{"size", "Max items to cache in RAM", util::ordinalCount(size_t{0}, size_t{100})} {}
+
+    struct Item {
+        std::chrono::system_clock::time_point used;
+        DataType data;
+    };
+
+    IntSizeTProperty size;
+    UnorderedStringMap<Item> cache;
+
+    void trim(size_t maxSize) {
+        while (cache.size() > maxSize) {
+            const auto it = std::ranges::min_element(
+                cache, std::ranges::less{}, [](const auto& pair) { return pair.second.used; });
+            cache.erase(it);
+        }
+    }
+
+    void add(std::string_view key, DataType data) {
+        if (size.get() == 0) return;
```

Comment:
I don't follow. Should be
```suggestion
        if (size.get() > 0 ) trim(size.get() - 1);
```